aiida/tools/importexport/dbimport/backends/django/__init__.py

```diff
@@ -559,24 +555,20 @@ def import_data_dj(
                         # is the case of the RETURN links of workflows/
                         # workchains. If it is not this case, then it is
                         # an error.
+                        from aiida.common.links import LinkType
                         existing_input = existing_input_links[out_id, link['label']]
 
-                        if link['type'] != LinkType.RETURN:
+                        if link['type'] != LinkType.RETURN.value:
```

Comment:
I'm not sure this is the correct logic (in the sense that there are other cases to take into account - e.g. CALL links).
I think we should use the logic that @sphuber implemented to check the validity and uniqueness constraints of link triples - @sphuber what do you think?

Reply:
I have not looked at the issue nor the solution in this PR, but yes, in general, the import code should not try and do the link validation itself. This is bound to become out of sync with the actual ORM implementation. I hope the import is done in a transaction, so I suggest to just try to import the nodes through the ORM and if a link is invalid it will raise the validation error in `Node.add_incoming` coming from `aiida.orm.utils.link.validate_link`. This can be caught and a more useful message can be formatted and printed. The only problem I can see is that going through the front-end ORM may be too slow for bulk imports.

Reply:
Makes sense. However, the biggest problem with export/import is how to make it general and keep up the speed (as hinted by @sphuber).
I suggest making another issue for this, since testing is necessary for the rewrite, which will take some time. The current quick-fix can then be reviewed as is, it should (according to the new test) fix the issue at hand, and the proper workflow can be implemented in a later PR?

Reply:
Probably this fix is enough for now for 1.0.
Maybe indeed this fix is critical for 1.0, and we don't have time to benchmark a better solution before 1.0.
So unless we see other problems with this PRI, I would be in favour of this temporary fix, adding a comment also in the code of why this has to be changed, and opening another issue

Reply:
Does this fix also now allow the scenario where a single `Data` node has two incoming `RETURN` links with the same label but from two different `WorkflowNodes`?

Reply:
> Does this fix also now allow the scenario where a single `Data` node has two incoming `RETURN` links with the same label but from two different `WorkflowNodes`?

I believe this is caught in the last `except KeyError`, where a new link is created .... Never mind, it isn't. It should be added as an `else` to the `if`-sentence that makes sure the link type is not a RETURN link. That should do it, I believe?

Reply:
Looking at the code it also seems that it is not even considering `CALL` links whatsoever. There may very well be quite some bugs with those as well.

Reply:
Again, to make it "function" somewhat, I can quick-fix it, however, the more I do that, the more I feel that we should simply bite the bullet and go through the ORM properly - making sure that this works properly and then prioritize optimization after 1.0?

Reply:
Concerning CALL links, this should be a check for existing output links, right? For RETURN links, the check is for existing input links.

Reply:
If we make certain assumptions about what can and can't be exported, then yes, but I think we should not make those assumptions. The only thing we should be checking is if adding the new link does not violate the rules. For the time being though, I guess one can only have the situation of additional incoming `CALL` links for an already existing node. The reverse, where an existing node, gets a new outgoing `CALL` link should not be possible.

Reply:
After discussing offline, we have found that the best thing to do now, will be to remove the existing logic and implement a "dumb" logic, i.e., go through each LinkType in turn and deal with them separately.
In this way, we make sure that everything works as intended for _this_ version.